test/lib/preboots/scheduler.test.js

```diff
@@ -0,0 +1,118 @@
+/* eslint max-nested-callbacks: 0 */
+/* eslint no-invalid-this: 0 */
+
+const Writer = require('../../mocks').Writer;
+const clone = require('clone');
+const assume = require('assume');
+const sinon = require('sinon');
+const packages = require('../../fixtures/packages.json');
+const heads = require('../../fixtures/heads');
+const path = require('path');
+
+describe('Scheduler', function () {
+  this.timeout(5E4);
+  let app = require('../../../lib');
+  let sandbox;
+  let scheduler;
+
+  before(function (done) {
+    app.start({
+      logger: {
+        level: 'critical'
+      },
+      ensure: true,
+      config: {
+        file: path.join(__dirname, '..', '..', 'config.json'),
+        overrides: {
+          http: 0,
+          scheduler: {
+            topic: 'testing'
+          }
+        }
+      }
+    }, function (err, application) {
+      if (err) return done(err);
+      app = application;
+      app.nsq = app.nsq || {};
+      app.nsq.writer = app.nsq.writer || new Writer();
+      app.scheduler.nsq = app.nsq;
+      scheduler = app.scheduler;
+      done();
+    });
+  });
+
+
+  after(function (done) {
+    app.close(done);
+  });
+
+  beforeEach(function () {
+    sandbox = sinon.sandbox.create();
+  });
+
+  afterEach(function () {
+    sandbox.restore();
+  });
+
+  it('setInterval should schedule a job that completes', function (done) {
+    sandbox.stub(scheduler.models.PackageCache, 'findAll').yieldsAsync(null, clone(packages));
+    const headStub = sandbox.stub(scheduler.models.BuildHead, 'findAll');
+    headStub.onCall(0).yieldsAsync(null, clone(heads.same['my-client-side-package']));
+    headStub.onCall(1).yieldsAsync(null, clone(heads.same['my-other-client-side-package']));
+
+    scheduler.once('scheduled', (err, counts) => {
+      const keys = Object.keys(counts);
+      assume(keys.length).equals(2);
```

Comment:
Update to use `assume@1.5.2` and you can use `assume(keys).length(2);` and it will give you a useful error if it's not 😉